src/auth.rs

```diff
@@ -1,3 +1,33 @@
+//! Client Authentication.
+//!
+//! Some RPC nodes will require authentication before requests can be sent to them.
+//!
+//! This is done by adding your api key to the request as a header.
+//!
+//! ## Example
+//!
+//! ```
+//! use near_jsonrpc_client::{JsonRpcClient, auth, methods};
+//! use near_primitives::types::{BlockReference, Finality};
+//!
+//! # #[tokio::main]
+//! # async fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let client = JsonRpcClient::connect("https://testnet.rpc.near.dev/")
```

Comment:
Let's make this `https://rpc.testnet.near.org`, going by what we use in Dev Console. https://console.pagoda.co/